layers/timeline_semaphore.c

```diff
@@ -80,6 +80,7 @@ struct object_map {
 struct instance_data {
     struct vulkan_vtable vtable;
     VkInstance instance;
+    uint32_t physicalDeviceCount;
```

Comment:
I could be missing something, but I wouldn't bother with storing the number of allocated devices.
Just query them all at destruction and remove them from the map.
The map should be able to deal with removing entries that were not inserted.

Reply:
There are two reasons,
1. Not all of the physicalDevices will be stored in global_map if  physicalDeviceCount specified by application is less than ICD driver reported. The physicalDeviceCount in instance_data here indicates how many physicalDevices are stored in global_map. The logic is clearer although hash table can handle the case which entry doesn't exist.
2. EnumeratePhysicalDevice would be called twice in InstanceDestroy if the physicalDeviceCount  is not stored.